mail_template_conditional_attachment/models/mail_template_conditional_attachment.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2019 Druidoo - Iván Todorovich <ivan.todorovich@druidoo.io>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, fields
+from odoo.tools import safe_eval
+import datetime
+import time
+import dateutil
+
+
+class MailTemplateConditionalAttachment(models.Model):
+    _name = 'mail.template.conditional.attachment'
+    _description = 'Mail Template Conditional Attachment'
+
+    name = fields.Char('Description', required=True)
+
+    mail_template_id = fields.Many2one(
+        'mail.template',
+        ondelete='cascade',
+        required=True,
+        index=True,
+    )
+
+    attachment_ids = fields.Many2many(
+        'ir.attachment',
+        string='Attachments',
+    )
+
+    model_name = fields.Char(related='mail_template_id.model_id.model')
+
+    filter_domain = fields.Char(
+        'Apply on',
+        help='If present, this condition must be satisfied '
+             'to include the attachments.',
+    )
+
+    def _get_eval_context(self):
+        return {
+            'datetime': datetime,
+            'dateutil': dateutil,
+            'time': time,
+            'uid': self.env.uid,
+            'user': self.env.user,
+        }
+
+    def _check_condition(self, res_id):
+        self.ensure_one()
+        if self.filter_domain:
```

Comment:
not a blocker but in such cases I usually do:
```py
if not self.filter_domain:
    return True
....
```
makes clear that you don't do anything in that case and avoid indentation of the whole code.
Feel free to ignore ;)